plugin-testing-core/src/main/java/com/palantir/gradle/plugintesting/GradleTestVersionsSource.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.plugintesting;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+@Target({ElementType.ANNOTATION_TYPE, ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Inherited
+@ArgumentsSource(GradleTestVersionsArgumentProvider.class)
```

Comment:
I don't have strong feelings (if anything, I like the separate class), but wondering whether this _could_ be `@MethodSource("com.palantir.gradle.plugintesting.GradleTestVersions#getGradleVersionsForTests")`?
```suggestion
@MethodSource("com.palantir.gradle.plugintesting.GradleTestVersions#getGradleVersionsForTests")
```

Reply:
Annotations don't work like this. An annotation isn't like a macro where it is simply expanded to the annotations applied to it - the annotations here apply to the `@GradleTestVersionsSource` macro itself.